File: easy/index-of-first-occurance.py
"""
 * Difficulty: easy
 * Find the Index of the First Occurrence in a String
 * difficulty: easy
 * https://leetcode.com/problems/find-the-index-of-the-first-occurrence-in-a-string/description/
"""
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        starter = needle[0]

        if(len(haystack) < len(needle)):
            return -1
        
        for idxI, i in enumerate(haystack):
            if i == starter:
                running_slice = ""
                for idxJ, j in enumerate(needle):
                    pointer = idxI + idxJ
                    if pointer >= len(haystack):
                        return -1
                    if haystack[pointer] == j:
                        running_slice += j
                if running_slice == needle:
                    return idxI
        return -1


# Run
solution = Solution()
res = solution.strStr("sadbutsad", "sad")
print(res)

File: easy/index-of-first-occurance.c
/**
 * Find the Index of the First Occurrence in a String
 * difficulty: easy
 * https://leetcode.com/problems/find-the-index-of-the-first-occurrence-in-a-string/description/
 **/

#include <stdio.h>
#include "helpers.h"


int strStr(char* haystack, char* needle) {
	// Get the length of the needle
	int lN = stringLength(needle);
	int lH = stringLength(haystack);

	if(lN > lH){
		return -1;
	}

	int x = 0;


	while(x < lH){
		char starter = haystack[x];

		int match = 0;
		if(starter == needle[0]){
			int i = 0;

			while(i < lN){
				if(needle[i] == haystack[x + i]){
					match += 1;
				}

				if(match == lN){
					return x;
				}
				i += 1;
			}
		}

		
		x += 1;
	}

	return -1;
}

int main(){
	int x = strStr("mississippi", "issipi");
	printf("%d\n", x);
	return 0;
}
